components/raftstore/src/store/peer.rs

```diff
@@ -2755,10 +2759,17 @@ where
                             .find(|x| {
                                 !self.disk_full_peers.has(x.get_id())
                                     && x.get_id() != self.peer.get_id()
+                                    && !self.down_peer_ids.contains(&x.get_id())
+                                    && matches!(x.get_role(), PeerRole::Voter)
```

Comment:
With joint consensus, there are also `IncomingVoter` and `DemotingVoter`, how about just `!matches!(x.get_role(), PeerRole::Learner)`

Reply:
OK,Changed.